oslabs-beta/watchmo
# Cli

Set up modular command line functionality using Yargs.

To add new function 'foo':
1.  Add function in new file ('foo.js' for example), 
2. Import into 'watchmo.js'  
3. Add to the '.' chain in 'watchmo.js' a .command("[command-name]", "[description]", [function definition]), 
e.g:
```
const { foo } = require('./foo');

argv
  //existing . chain
  .command('foo', 'does a thing', foo)
```

Added functionality for 'init' command using node's fs packages. Init will create a new directory in the project file structure '/watchmo' containing a directory '/snapshots' and '/config.json'. The config file keeps track of the queries to watch and the GQL endpoint.

The other CLI functions (watch, mo, default) are still in development. Currently, 'watch' successfully grabs the endpoint and queries from the config file using yargs' '.config' method.

TO TEST:
We are still waiting on Andrew to let us know how to make CLI commands easily accessible upon npm install. In the meanwhile:

1. In the watchmo project directory, run from the terminal `npm link`. You may need to run this as a superuser, i.e. `sudo npm link`. This should add the watchmo CLI commands to your terminal commands. (`npm unlink` in the watchmo project directory removes these new commands)
2. Create a new project directory
3. cd into new project directory
4. run `watchmo init` and see if it creates the appropriate file structure
4*. run `watchmo --help` to see other commands.
*play around with these if you want
